express/scripts/scripts.js

```diff
@@ -1065,78 +1076,213 @@ export function checkTesting() {
   return (getMeta('testing').toLowerCase() === 'on');
 }
 
-async function decorateTesting() {
-  let runTest = true;
-  // let reason = '';
-  const usp = new URLSearchParams(window.location.search);
-  const martech = usp.get('martech');
-  if ((checkTesting() && (martech !== 'off') && (martech !== 'delay')) || martech === 'rush') {
-    // eslint-disable-next-line no-console
-    console.log('rushing martech');
-    loadScript('/express/scripts/instrument.js', null, 'module');
-  }
+/**
+ * Sanitizes a name for use as class name.
+ * @param {*} name The unsanitized name
+ * @returns {string} The class name
+ */
+export function toCamelCase(name) {
+  return toClassName(name).replace(/-([a-z])/g, (g) => g[1].toUpperCase());
+}
+
+export function getExperiment() {
+  let experiment = getMeta('experiment').toLowerCase();
 
-  if (!window.location.host.includes('adobe.com')) {
-    runTest = false;
+  if (!window.location.host.includes('adobe.com') && !window.location.host.includes('.hlx.live')) {
+    experiment = '';
     // reason = 'not prod host';
   }
   if (window.location.hash) {
-    runTest = false;
+    experiment = '';
     // reason = 'suppressed by #';
   }
-  if (window.location.search === '?test') {
-    runTest = true;
-  }
+
   if (navigator.userAgent.match(/bot|crawl|spider/i)) {
-    runTest = false;
+    experiment = '';
     // reason = 'bot detected';
   }
 
-  if (runTest) {
-    let $testTable;
-    document.querySelectorAll('table th').forEach(($th) => {
-      if ($th.textContent.toLowerCase().trim() === 'a/b test') {
-        $testTable = $th.closest('table');
-      }
+  const usp = new URLSearchParams(window.location.search);
+  if (usp.has('experiment')) {
+    [experiment] = usp.get('experiment').split('/');
+  }
+
+  return experiment;
+}
+
+/**
+ * Gets experiment config
```

Comment:
Would be great to describe how the json file should look like... Looking at the function code, there are some expectations on the format that need to be reversed-engineered from the 2 forEach.

Reply:
and/or add some tests. At least it will illustrate the various expected json manifest

Reply:
done